### PR TITLE
Add os_project_facts module

### PR DESCRIPTION
**Pull request by rcarrillocruz**
##### ISSUE TYPE

- New Module Pull Request
##### COMPONENT NAME

os_projects_facts
##### ANSIBLE VERSION

```
ansible 2.1.0 (devel cca084c89d) last updated 2016/04/05 14:55:00 (GMT +000)
  lib/ansible/modules/core: (detached HEAD cf01087a30) last updated 2016/04/05 14:55:07 (GMT +000)
  lib/ansible/modules/extras: (os_project_facts f1dc4a43e9) last updated 2016/04/05 17:14:46 (GMT +000)
  config file = 
  configured module search path = Default w/o overrides
```
##### SUMMARY

This module gathers facts about OpenStack projects and returns them to the user.

### CODE REVIEW

File: cloud/openstack/os_project_facts.py [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+            filters['domain_id'] = domain
+
+        cloud = shade.openstack_cloud(**module.params)
+        projects = cloud.search_projects(name,
+                                        filters)
+        module.exit_json(changed=False, ansible_facts=dict(
+            openstack_projects=projects))
+
+    except shade.OpenStackCloudException as e:
+        module.fail_json(msg=e.message, extra_data=e.extra_data)
```
Comment (Shrews) [inline cloud/openstack/os_project_facts.py]: This should be:

  `module.fail_json(msg=str(e))`

File: cloud/openstack/os_project_facts.py [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+   domain:
+     description:
+        - Name or ID of the domain containing the project if the cloud supports domains
+     required: false
+     default: None
+   filters:
+     description:
+        - A dictionary of meta data to use for further filtering.  Elements of
+          this dictionary may be additional dictionaries.
+     required: false
```
Comment (Shrews) [inline cloud/openstack/os_project_facts.py:50]: required=false needs to document the default value, per http://docs.ansible.com/ansible/developing_modules.html#module-checklist

File: cloud/openstack/os_project_facts.py [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+                except:
+                    # Ok, let's hope the user is non-admin and passing a sane id
+                    pass
+
+            if not filters:
+                filters = {}
+
+            filters['domain_id'] = domain
+
+        cloud = shade.openstack_cloud(**module.params)
```
Comment (Shrews) [inline cloud/openstack/os_project_facts.py]: This doesn't seem necessary if you already have an opcloud defined. Couldn't we just reuse that one?

File: cloud/openstack/os_project_facts.py [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+            description: Name given to the project.
+            returned: success
+            type: string
+        description:
+            description: Description of the project
+            returned: success
+            type: string
+        enabled:
+            description: Flag to indicate if the project is enabled
+            return: success
```
Comment (Shrews) [inline cloud/openstack/os_project_facts.py]: s/return/returned/

File: cloud/openstack/os_project_facts.py [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+            type: bool
+        domain_id:
+            description: Domain ID containing the project
+            return: success
```
Comment (Shrews) [inline cloud/openstack/os_project_facts.py]: s/return/returned/

**Comment by Shrews** [comment]
I prefer it be named os_project_facts since we have an os_project module (to keep the names similar).

File: cloud/openstack/os_project_facts.py (hunk 3 above)
Comment (rcarrillocruz) [inline cloud/openstack/os_project_facts.py]: Hmm, yeah, OperatorCloud subclasses OpenStackCloud.
Which makes me think I could later refactor os_project (I did some copy-pasta from it here).

**Comment by rcarrillocruz** [comment]
ready_for_review

**Comment by gregdek** [comment]
Thanks @rcarrillocruz. To the current maintainers, @emonty, @shrews, @juliakreger, @j2sol, @rcarrillocruz please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit' or 'needs_revision' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

File: cloud/openstack/os_project_facts.py [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+extends_documentation_fragment: openstack
+version_added: "2.1"
+author: "Ricardo Carrillo Cruz (@rcarrillocruz)"
+description:
+    - Retrieve facts about a one or more OpenStack projects
+requirements:
+    - "python >= 2.6"
+    - "shade"
+options:
+   project:
```
Comment (Shrews) [inline cloud/openstack/os_project_facts.py]: This does not match the parameter name ('name').

File: cloud/openstack/os_project_facts.py [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+        description:
+            description: Description of the project
+            returned: success
+            type: string
+        enabled:
+            description: Flag to indicate if the project is enabled
+            returned: success
+            type: bool
+        domain_id:
+            description: Domain ID containing the project
```
Comment (Shrews) [inline cloud/openstack/os_project_facts.py]: Probably should mention that domain_id is returned only with keystone v3.

File: cloud/openstack/os_project_facts.py [hunk 8]
```diff
@@ -0,0 +1,164 @@
...
+                else:
+                    module.fail_json(msg='Domain name or ID does not exist')
+
+            if not filters:
+                filters = {}
+
+            filters['domain_id'] = domain
+
+        projects = opcloud.search_projects(name,
+                                        filters)
```
Comment (Shrews) [inline cloud/openstack/os_project_facts.py]: Weird spacing here in search_project() params. No need for the extra \n.

**Comment by rcarrillocruz** [comment]
ready_for_review

**Comment by gregdek** [comment]
Thanks @rcarrillocruz. To the current maintainers, @emonty, @shrews, @juliakreger, @j2sol, @rcarrillocruz please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit' or 'needs_revision' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

**Pull request merged**